src/PhpParser/Node/Value/TernaryBracketWrapper.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\PhpParser\Node\Value;
+
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\Name;
+use Rector\Core\PhpParser\Printer\BetterStandardPrinter;
+
+final class TernaryBracketWrapper
+{
+    public function __construct(private BetterStandardPrinter $betterStandardPrinter)
+    {
+    }
+
+    public function getWrappedTernaryConstFetch(Ternary $ternary): ConstFetch
+    {
+        return new ConstFetch(new Name('(' . $this->betterStandardPrinter->print($ternary) . ')'));
+    }
+}
```

Comment:
This is rather hacking node contents. The `(` `)` actually are not part of the `Name`.

More clean way is to use attribute and then extend `BetterStandardPrinter` to print this node with the attribute correctly. E.g. `Array_` has `setAttribute('kind', ...)` value that handles 2 prints:

```php
array()
[]
```

Here in this file that means something like this:

```php
$ternary->setAttribute('kind', 'wrapped_with_brackets');
```

Then add a new overload method to `BetterStandardPrinter`:

```php
protected function pExpr_Ternary(Expr\Ternary $node)
{
    // see the parent method for inspiration
    // ...
}
```

Reply:
I will try.

Reply:
implemented https://github.com/rectorphp/rector-src/pull/397/commits/92d341ae4778610f7f0b93295881065c4eb99d9d#diff-35b384f5686672e9b726b991a0c45dd8c18cfc8b365953489bf95a39db82205fR367-R375